### DISCUSSION TITLE
How can we store blob data?

### DISCUSSION THREAD

**Discussion by IAmMythoRED** [prisma/prisma#21689] [Q&A] [answered]
### Question

Is there any way to store a blop data ? is there any data type? 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @BotBuddies 👋 

Thank you for raising this question.

It’s usually a good practice to store Images or videos in a Blob Storage and images or videos should not directly be stored in the database, so you can use something like [AWS S3](https://aws.amazon.com/s3/) or [Cloudinary](https://cloudinary.com/) to store the image/video and store the bucket path in your database.

Here’s a working [example](https://cloudinary.com/blog/upload-and-display-images-using-prisma) of how you could achieve it.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by dopry** [reply]
The question didn't say anything about images or videos. There are plenty of use cases for storing blobs in a database.  This response doesn't answer how to use a blob in prisma.

**Comment by FlorianWendelborn** [comment]
There is a `Bytes` data type in Prisma:
- <https://www.prisma.io/docs/orm/v6/prisma-client/special-fields-and-types#working-with-bytes>
- <https://www.prisma.io/docs/orm/reference/prisma-schema-reference#bytes>

Please note however, that it is not _usually_ advisable to store bytes directly in the database, especially if there will be a lot of them. For bulk storage, an S3-compatible storage API would _usually_ be preferable.
